src/early-errors.js

```diff
@@ -165,18 +165,20 @@ export class EarlyErrorChecker extends MonoidalReducer {
   }
 
   reduceCatchClause(node, { binding, body }) {
-    binding = binding.observeLexicalDeclaration();
-    binding = binding.enforceDuplicateLexicallyDeclaredNames(DUPLICATE_BINDING);
-    binding = binding.enforceConflictingLexicallyDeclaredNames(body.previousLexicallyDeclaredNames, DUPLICATE_BINDING);
-    binding.lexicallyDeclaredNames.forEachEntry((nodes, bindingName) => {
-      if (body.varDeclaredNames.has(bindingName)) {
-        body.varDeclaredNames.get(bindingName).forEach(conflictingNode => {
-          if (body.forOfVarDeclaredNames.indexOf(conflictingNode) >= 0) {
-            binding = binding.addError(DUPLICATE_BINDING(conflictingNode));
-          }
-        });
-      }
-    });
+    if (binding != null) {
+      binding = binding.observeLexicalDeclaration();
+      binding = binding.enforceDuplicateLexicallyDeclaredNames(DUPLICATE_BINDING);
+      binding = binding.enforceConflictingLexicallyDeclaredNames(body.previousLexicallyDeclaredNames, DUPLICATE_BINDING);
+      binding.lexicallyDeclaredNames.forEachEntry((nodes, bindingName) => {
+        if (body.varDeclaredNames.has(bindingName)) {
+          body.varDeclaredNames.get(bindingName).forEach(conflictingNode => {
+            if (body.forOfVarDeclaredNames.indexOf(conflictingNode) >= 0) {
```

Comment:
https://github.com/tc39/ecma262/commit/8a16cb8 removed this error, and was included in es2019. The `forOfVarDeclaredNames` infrastructure isn't necessary any more.

Specifically, prior to that commit

```js
try {
} catch (e) {
  for (var e of []);
}
```
was illegal, and now (in es2019 and up) it's legal. We should add a test for it, too.